x-pack/plugin/sql/qa/server/src/main/java/org/elasticsearch/xpack/sql/qa/cli/SelectTestCase.java

```diff
@@ -37,4 +37,45 @@ public void testSelectWithWhere() throws IOException {
         assertThat(readLine(), RegexMatcher.matches("\\s*2\\s*\\|\\s*test_value2\\s*"));
         assertEquals("", readLine());
     }
+
+    public void testLeniency() throws IOException {
+        EmbeddedCli cliWithLeniency = new EmbeddedCli(
+            CliIntegrationTestCase.elasticsearchAddress(),
+            true,
+            securityConfig(),
+            new String[] { "-lenient", "true" }
+        );
+
+        try {
+            index("test", body -> body.field("name", "foo").field("tags", new String[] { "bar", "bar" }));
+
+            assertThat(cliWithLeniency.command("SELECT * FROM test"), RegexMatcher.matches("\\s*name\\s*\\|\\s*tags\\s*"));
+            assertThat(cliWithLeniency.readLine(), containsString("----------"));
+            assertThat(cliWithLeniency.readLine(), RegexMatcher.matches("\\s*foo\\s*\\|\\s*bar\\s*"));
+            assertEquals("", cliWithLeniency.readLine());
+        } finally {
+            cliWithLeniency.close();
+        }
+    }
+
+    public void testLeniency2() throws IOException {
+        EmbeddedCli cliWithLeniency = new EmbeddedCli(
+            CliIntegrationTestCase.elasticsearchAddress(),
+            true,
+            securityConfig(),
+            new String[] { "-l", "true" }
+        );
+
+        try {
+            index("test", body -> body.field("name", "foo").field("tags", new String[] { "bar", "bar" }));
+
+            assertThat(cliWithLeniency.command("SELECT * FROM test"), RegexMatcher.matches("\\s*name\\s*\\|\\s*tags\\s*"));
+            assertThat(cliWithLeniency.readLine(), containsString("----------"));
+            assertThat(cliWithLeniency.readLine(), RegexMatcher.matches("\\s*foo\\s*\\|\\s*bar\\s*"));
+            assertEquals("", cliWithLeniency.readLine());
+        } finally {
+            cliWithLeniency.close();
+        }
+    }
+
```

Comment:
Instead of duplicating this code, use a single test method where for the arguments list you use a `.randomFrom` and use the two variants as values to choose from.

Reply:
Good point, fixing